### ISSUE TITLE
why the loss is rising after few epoch? can't understand it

### ISSUE THREAD

**Issue by xmuszq** [deepinsight/insightface#377]
I trained with Resnet 100 with new cleaned MS data @ 4GPU @128 batchsize/GPU, lr=0.1 at first, from the loss it looks weird to me:
1. the sudden large noise in the loss (the left red circle)
2. the loss is rising (in the right red circle)

Anyone can give me some suggestions?
![11](https://user-images.githubusercontent.com/15931069/45975149-6b2eb400-c011-11e8-96cc-4aaa73d45444.png)

**Comment by ltcs11** [comment]
i have met with problem 2 before **(but in tensorflow, just for reference)**
you can see from not only the right red circle but also the whole training process, that the loss would change cyclicity in every epoch
those situation happens because of the data shuffle method. Since that the dataset is too large, data shuffle module has to use data buffer or FIFO .etc to shuffle only part of the whole dataset in pipeline.
So when it comes to the end of every epoch, the last part of the original dataset would  be easier to accumulate in sequences(Since they are also the last part to get in the sequences). So if your dataset is in order, their label would be easier to be the same. This  would make the network get over-fitted(if you feed the network with similar input), as you can see from the right red circle that the loss decrease fast at the beginning. And then the loss would rising slowly as the new epoch came, which means the input data become less similar and more random.
If you want to solve this problem, the best idea is to shuffle the whole dataset totally before.

**Comment by xmuszq** [comment]
@ltcs11    
Thank you so much for your information! I will try your through and see. 

Best

**Comment by zzzzz314314** [comment]
Hello,
I met a similar problem, except that my training data is pre-shuffled throughly before the training starts. However, the same case persists across many trials. I also posted an detailed [issue](https://github.com/deepinsight/insightface/issues/2059) for this. Please give some advice on this if you have experience. Many thanks.
